authors/apps/articles/views.py

```diff
@@ -131,3 +135,88 @@ def put(self, request, slug):
                                            context={'request': request},
                                            partial=True)
         return Response(serializer.data, status.HTTP_200_OK)
+
+
+class CommentDetailsView(RetrieveUpdateAPIView):
+    """This class handles the http GET and PUT requests."""
+    permission_classes = (IsAuthorOrReadOnly,)
+    queryset = Comment.objects.filter(is_deleted=False)
+    serializer_class = Comment
+    lookup_field = 'id'
+
+
+class CommentDelete(UpdateAPIView):
+
+    """This class Deletes Comments created by the a particular user"""
+    permission_classes = (IsAuthorOrReadOnly,)
+    queryset = Comment.objects.filter(is_deleted=False)
+    serializer_class = DeleteCommentSerializer
+    lookup_field = 'id'
+
+
+class CommentDetailsView(RetrieveUpdateAPIView):
+    """This class handles the http GET and PUT requests."""
+    permission_classes = (IsAuthorOrReadOnly,)
+    queryset = Comment.objects.filter(is_deleted=False)
+    serializer_class = CommentSerializer
+    lookup_field = 'id'
+
+
+class CreateComment(generics.ListCreateAPIView):
+    """This class Creates comments."""
+    serializer_class = CommentSerializer
+    queryset = Comment.objects.filter(is_deleted=False)
+    permission_classes = (IsAuthenticated, )
+
+    def post(self, request, slug, *args, **kwargs):
+        article = Article.objects.filter(slug=slug, is_deleted=False).first()
+        serializer_context = {
+            'request': request,
+            'author': request.user,
+            'article': get_object_or_404(Article, slug=self.kwargs["slug"])
+        }
+        print((request.data))
+        serializer = CommentSerializer(data=request.data, context=serializer_context)
+        if serializer.is_valid():
+            serializer.save(author=request.user, article_id=article.id)
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class CommentsRetrieveUpdateDestroy(RetrieveUpdateDestroyAPIView, CreateAPIView):
+    serializer_class = CommentSerializer
+    permission_classes = (IsAuthenticated,)
+    queryset = Comment.objects.filter(is_deleted=False)
+    lookup_field = 'id'
+
+    def create(self, request, slug, id):
+        """Method for creating a child comment on parent comment."""
+
+        context = super(CommentsRetrieveUpdateDestroy,
+                        self).get_serializer_context()
+
+        comment = Comment.objects.filter(id=id, is_deleted=True).first()
```

Comment:
ditto

Reply:
We've also checked this with @d-kahara but am getting this error Comment matching query does not exist.